product-service/handlers/catalogBatchProcess.ts

```diff
@@ -0,0 +1,63 @@
+import { SQSHandler } from "aws-lambda";
+import AWS from "aws-sdk";
+import "source-map-support/register";
+import {    
+    BUCKET_REGION as REGION,    
+} from "../../import-service/constants";
+import { createProduct } from "../services";
+import { CSVProduct } from "../models/db";
+
+export const catalogBatchProcess: SQSHandler = async (event, _context, cb) => {
+    const sns = new AWS.SNS({ region: REGION });
+    const products = [];
+
+    for (const record of event.Records) {
+        try {
+            const {
+                count,
+                description,
+                imageurl,
+                price,
+                title,
+            }: CSVProduct = JSON.parse(record.body);
+
+            if (!count || !description || !price || !title) {
+                throw new Error(
+                    "invalid data record"
+                );
+            }
+
+            const product = {
+                description,
+                imageUrl: imageurl,
+                price: +price,
+                title,
+                count: +count,
+            };
+
+            products.push(product)
+            console.log('Creating product', JSON.stringify(product))
+
+            await createProduct(product);
+
+        } catch {
+            console.error(
+                "catalogBatchProcess found invalid JSON record:",
+                record.body
+            );
+            return;
```

Comment:
Return from the for...of cycle is not working as you expect.

Reply:
+